kyuubi-server/src/main/scala/org/apache/kyuubi/operation/BatchJobSubmission.scala

```diff
@@ -43,12 +43,14 @@ class BatchJobSubmission(session: KyuubiBatchSessionImpl, batchRequest: BatchReq
 
   private lazy val _operationLog = OperationLog.createOperationLog(session, getHandle)
 
-  private var builder: ProcBuilder = _
+  private val applicationManager =
+    session.sessionManager.asInstanceOf[KyuubiSessionManager].applicationManager
 
-  @volatile
-  private[kyuubi] var appIdAndUrl: Option[(String, String)] = None
+  private var builder: ProcBuilder = _
 
-  private var resultFetched: Boolean = _
+  private[kyuubi] def currentApplicationState: Option[Map[String, String]] = {
+    applicationManager.getApplicationInfo(builder.clusterManager(), statementId)
```

Comment:
Here the statementId is the Operation handle identifier not KyuubiBatchSession handle identifier.

I think that it is better to use KyuubiBatchSession handle identifier, so that customers can also check batch job application by batch id.

Reply:
updated